### PR TITLE
Updating the client_config_resource

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
Per this bug: https://chefio.atlassian.net/browse/CHEF-7885, we had previously taken a fix to correct the issue. However, that fix only addressed a problem with the Ohai property and not the policy_persist_run_list property. This fix updates that. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by Stromweld** [comment]
you also need to add to the property `default: false` https://github.com/chef/chef/blob/main/lib/chef/resource/chef_client_config.rb#L196

File: lib/chef/resource/support/client.erb [hunk 1]
```diff
@@ -16,12 +16,12 @@
       @policy_group
       @policy_name
       @rubygems_url
-      @ssl_verify_mode
-      @policy_persist_run_list).each do |prop| -%>
+      @ssl_verify_mode).each do |prop| -%>
 <% next if instance_variable_get(prop).nil? || instance_variable_get(prop).empty? -%>
 <%=prop.delete_prefix("@") %> <%= instance_variable_get(prop).inspect %>
 <% end -%>
```
Comment (tpowell-progress) [inline lib/chef/resource/support/client.erb:22]:
```suggestion
<% end -%>
<%# boolean properties are neither .nil? nor respond to .empty? so they are included below %>
```

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
Add the comment in case further issues come up or someone decides to refactor.

**Review by tpowell-progress — APPROVED** [review]

File: lib/chef/resource/support/client.erb (hunk 1 above)
Comment (johnmccrae) [reply lib/chef/resource/support/client.erb:22]: done

**Pull request merged**